library/include/borealis/views/image.hpp

```diff
@@ -95,6 +95,23 @@ class Image : public View
      */
     void setImageFromFile(std::string path);
 
+    /**
+     * Sets the image from a memory buffer. The data read has to be
+     * the same as if the image would've been read from a file.
+     * 
+     * See NanoVG documentation and the Image class documentation for the
+     * list of supported image formats.
+     */
+    void setImageFromMemory(unsigned char* data, int numData);
+
+    /**
+     * Sets the image from a memory buffer. The data should be a
+     * raw four-channel (RGBA) pixel array of the size width*height*4.
+     * 
+     * See the example implementation dynamic_image in the demo.
+     */
+    void setImageFromRawData(const unsigned char* data, int width, int height);
```

Comment:
"raw data" can be anything: RGB, RGBA, YUV, ABRG...

I would call it "setImageFromRGBA" rather than "raw data", and most importantly specify the _full_ pixel format in the documentation (RGBA8888 if I'm not mistaken?). This is better than asking for a "width * height * 4" array.